### PR TITLE
#5384 add GetCategorisations function

### PR DESCRIPTION
**Pull request by andre-urbani**
### What

Add dp-api-clients-go function for GET /population-types/{population-type}/dimensions/{dimension}/categorisations
Trello - https://trello.com/c/5Ltsbl56/5834-add-dp-api-clients-go-function-for-get-population-types-population-type-dimensions-dimension-categorisations

### How to review

Confirm changes match requested changes in Trello ticket

### Who can review

Anyone

### CODE REVIEW

File: cantabular/contract.go [hunk 1]
```diff
@@ -158,6 +158,18 @@ type GetParentsResponse struct {
...
+	PaginationParams
+	Dataset  string
+	Variable string
+}
+
+// GetCategorisationsResponse is the response body for the GetCategorisations query
+type GetCategorisationsResponse struct {
+	Dataset gql.Dataset `json:"dataset"`
+}
+
```
Comment (PauloPortugal) [inline cantabular/contract.go:172]: Do we need to know the `totalCount` and this `count` batch?

File: cantabular/dimensions.go [hunk 2]
```diff
@@ -422,3 +422,34 @@ func (c *Client) GetGeographyBatchProcess(ctx context.Context, datasetID string,
...
+			errors.New("error(s) returned by graphQL query"),
+			resp.Errors[0].StatusCode(),
+			log.Data{
+				"request": req,
+				"errors":  resp.Errors,
+			},
+		)
+	}
+
+	return &resp.Data, nil
```
Comment (PauloPortugal) [inline cantabular/dimensions.go:454]: Do we need to add some `PaginationParams` here as well?
@franmoore05 probably you can advise us here :slightly_smiling_face:

File: cantabular/queries.go [hunk 3]
```diff
@@ -251,6 +251,30 @@ query ($dataset: String!, $variables: [String!]!, $limit: Int!, $offset: Int) {
...
+                    label
+				    code
+                	}
+           		}
+			  name
+			  label
+       	      }
+		    }
+		  }
+		}
```
Comment (PauloPortugal) [inline cantabular/queries.go:273]: Could you please correct the indentation here?

File: cantabular/contract.go (hunk 1 above)
Comment (andre-urbani) [reply cantabular/contract.go]: Yes you're right. Now added in response

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**